AISKU/src/ApplicationInsightsDeprecated.ts

```diff
@@ -0,0 +1,317 @@
+        throw new Error("downloadAndSetup not implemented in web SKU");
+    }
+
+    // note: these are split into methods to enable unit tests
+    public loadAppInsights() {
+
+        // initialize global instance of appInsights
+        //var appInsights = new Microsoft.ApplicationInsights.AppInsights(this.config);
+
+        // implement legacy version of trackPageView for 0.10<
+        if (this.config["iKey"]) {
+            var originalTrackPageView = this.trackPageView;
+            this.trackPageView = (pagePath?: string, properties?: Object, measurements?: Object) => {
+                originalTrackPageView.apply(this, [null, pagePath, properties, measurements]);
+            }
+        }
+
+        // implement legacy pageView interface if it is present in the snippet
+        var legacyPageView = "logPageView";
+        if (typeof this.snippet[legacyPageView] === "function") {
+            this[legacyPageView] = (pagePath?: string, properties?: Object, measurements?: Object) => {
+                this.trackPageView(null, pagePath, properties, measurements);
+            }
+        }
+
+        // implement legacy event interface if it is present in the snippet
+        var legacyEvent = "logEvent";
+        if (typeof this.snippet[legacyEvent] === "function") {
+            this[legacyEvent] = (name: string, props?: Object, measurements?: Object) => {
+                this.trackEvent(name, props, measurements);
+            }
+        }
+
+        return this;
+    }
+
+    private static getDefaultConfig(config?: any): any {
+        if (!config) {
+            config = <any>{};
+        }
+
+        // set default values
+        config.endpointUrl = config.endpointUrl || "https://dc.services.visualstudio.com/v2/track";
+        config.sessionRenewalMs = 30 * 60 * 1000;
+        config.sessionExpirationMs = 24 * 60 * 60 * 1000;
+        config.maxBatchSizeInBytes = config.maxBatchSizeInBytes > 0 ? config.maxBatchSizeInBytes : 102400; // 100kb
+        config.maxBatchInterval = !isNaN(config.maxBatchInterval) ? config.maxBatchInterval : 15000;
+        config.enableDebug = Util.stringToBoolOrDefault(config.enableDebug);
+        config.disableExceptionTracking = Util.stringToBoolOrDefault(config.disableExceptionTracking);
+        config.disableTelemetry = Util.stringToBoolOrDefault(config.disableTelemetry);
+        config.verboseLogging = Util.stringToBoolOrDefault(config.verboseLogging);
+        config.emitLineDelimitedJson = Util.stringToBoolOrDefault(config.emitLineDelimitedJson);
+        config.diagnosticLogInterval = config.diagnosticLogInterval || 10000;
+        config.autoTrackPageVisitTime = Util.stringToBoolOrDefault(config.autoTrackPageVisitTime);
+
+        if (isNaN(config.samplingPercentage) || config.samplingPercentage <= 0 || config.samplingPercentage >= 100) {
+            config.samplingPercentage = 100;
+        }
+
+        config.disableAjaxTracking = Util.stringToBoolOrDefault(config.disableAjaxTracking);
```

Comment:
also need disableFetchTracking (and enableOldTags?)

Reply:
True, let me see whats a good solution for properties in this scenario.

Reply:
I'll need to essentially change properties package to support all old or all new. I'll do that in next PR along with support for addTelemetryInitializer.